Charging_Current.cpp

```diff
@@ -0,0 +1,42 @@
+#include "Charging_Current.h"
+using namespace std;
+
+void generateRangeAndPrintOutput(std::vector<int> input_sample)
+{
+  uint8_t i;
+  int count = 0;
+  std::vector<int> range;
+  std::cout << "|   RANGE   |   READINGS   |" << std::endl;
+  std::cout << "----------------------------------------" << std::endl;
+
+  if (!input_sample.empty())
+  {
+    for (i = 0; i < input_sample.size(); ++i)
+    {
+
+      if (count == 0)
+      {
+        range.push_back(input_sample[i]);
+      }
+
+      if (input_sample[i + 1] == input_sample[i] + 1)
```

Comment:
i+1 for the last element will try to access element beyond the array size. This might lead to a inconsistent behavior. The accessed value could be a garbage value or could cause fault exception.
The design should handle such inconsistency.

Reply:
hello , Im not able to find any solution for this issue ... any suggestions or hints ? Or Do I need to change the full implementation?

Reply:
If you want to use the same condition and indexes, I suggest to keep the for loop range to 0 to input_sample size -1. in this way the final pass of the loop will be consider last and last but one element.

Reply:
Hello, I tried this solution before also ... but what happens is ,it wont consider the ranges from 10 onwards ...as in the test result it can be seen that it only gives correct result till range 7-9. Maybe I'm missing something ,can you give some solution to this?!!